sleigh/sleigh-runtime/src/decoder.rs

```diff
@@ -17,6 +17,9 @@ pub struct Decoder {
     /// The local context value.
     pub(crate) context: u64,
 
+    /// The future context modifications (Addr, Field, Val).
+    future_context_mods: Vec<(u64, Field, i64)>,
```

Comment:
I think it's probably better to use `HashMap<u64, Vec<Field, i64>>` here, to avoid a linear search if there are a lot of context modifications?

Otherwise, if there's something I'm missing, maybe add a comment explaining things here?

Reply:
Actually, checking the implementation below, it looks like only one future context mod is allowed per field?

I couldn't find anything about this in the SLEIGH reference, as far as I could tell, multiple global sets for the same location should be allowed.